src/stack.rs

```diff
@@ -1,14 +1,123 @@
-// Alex Ozdemir <aozdemir@hmc.edu> // <- Your name should replace this line!
-// Starter code for HMC's MemorySafe, week 1
+// Luis Viornery <lviornery@hmc.edu>
+// Significant help from Ross Mawhorter
+// Code for HMC's MemorySafe, week 1
 //
 // The implementation of SinglyLinkedList
 
 use Stack;
+use std::mem;
+
+
+pub struct Node<T> {
+//We have a node of the struct with a data and a next_node member
+//all Nodes in a lost should be Boxed
+    data: T,
+    //next_node may be None, or it may be a boxed node
+    next_node: Option<Box<Node<T>>>,
+}
 
 pub struct SinglyLinkedList<T> {
-    data: T // Probably not what you want
+//The singlylinkedlist has no data member, and it has a length
+    top_node: Option<Box<Node<T>>>,
+    length: usize,
 }
 
 impl<T: Eq> Stack<T> for SinglyLinkedList<T> {
-    // ????
+    //make a new list with no top node and 0 length
+    fn new() -> Self {
+        SinglyLinkedList{
+            top_node: None,
+            length: 0,
+        }
+    }
+    fn push_front(&mut self, item: T) {
+    //add 1 to length, then create a new boxed node
+        self.length += 1;
+        let mut new_box = Box::new(Node {
+                data: item,
+                next_node: None,
+            });
+        //then make self.topnode point to None
+        //and the new box's next_node point to the previous top_node
+        mem::swap(&mut self.top_node, &mut new_box.next_node);
+        //then make the top node point to the new box
+        //(moving gives the list ownership of the box)
+        self.top_node = Some(new_box);
+    }
+    fn pop_front(&mut self) -> Option<T> {
+    //replace the top element of the list with the second element if there is one
+        //old_top is the Option<Box<Node<T>>> that is the old top_node
+        let old_top = mem::replace(&mut self.top_node, None);
+        match old_top {
+            //If there was no old top node, return nothing
+            None => None,
+            //If there was...
+            Some(boxed_node) => {
+                //dereference the box
+                let d_b_n = *boxed_node; //defererenced_boxed_node
```

Comment:
Nice! It turns out that doing this is super important for allowing you to move d_b_n's fields around separately.